rlay-client/src/rpc/mod.rs

```diff
@@ -99,7 +99,15 @@ async fn run_rpc(
         let sync_state = sync_state.clone();
         async {
             Ok::<_, GenericError>(service_fn(move |req| {
-                handle_jsonrpc(full_config.clone(), sync_state.clone(), req)
+                match (req.method(), req.uri().path()) {
+                    (&Method::GET, "/health") => {
+                        http_get_health().boxed()
```

Comment:
I think you might be able to just `.await` on the futures instead of using `.boxed()`

Reply:
That opens up a problem of not being await to run await in non-async blocks which gives you two options (1) make the closure async, which is unstable though and throws and (2) make an async block, but that opens a huge can of worms that I was unable to figure out.

Let me know if you see another options

Reply:
Oh, didn't see the closure. Yeah, then it's probably not worth it.